common/src/main/java/org/conscrypt/ClientSessionContext.java

```diff
@@ -86,11 +92,20 @@ NativeSslSession getCachedSession(String hostName, int port, SSLParametersImpl s
             return null;
         }
 
+        if (session.isSingleUse()) {
+            removeSession(session);
+        }
```

Comment:
Sorry, I missed this the first time: this will allow a session to be used multiple times if two connections start in parallel on two different threads. Do you all care about this? (It does make it likely that the two connections will go over the same network interface, at which point they're correlated by IPs anyway, but making them actually single-use means you don't have to think about it.)

Reply:
Good catch, made this synchronized, which should eliminate that possibility.